flake.nix

```
@@ -0,0 +1,98 @@
+# SPDX-FileCopyrightText: 2022 Serokell <https://serokell.io>
+# SPDX-License-Identifier: MPL-2.0
+
+{
+  description = "The xrefcheck flake";
+
+  nixConfig.flake-registry = "https://github.com/serokell/flake-registry/raw/master/flake-registry.json";
+
+  outputs = { self, flake-utils, haskell-nix, serokell-nix, ... }:
+    flake-utils.lib.eachSystem [ "x86_64-linux" ] (system:
+    let
+      pkgs = haskell-nix.legacyPackages.${system}.extend serokell-nix.overlay;
+
+      # source with gitignored files filtered out
+      projectSrc = pkgs.haskell-nix.haskellLib.cleanGit {
+        src = ./.;
+        name = "xrefcheck";
+        keepGitDir = true;
+      };
```

Flakes filter gitignored files automatically

```suggestion
      projectSrc = ./.;
```